src/storage/keyvalue/WrappedExpiringStorage.ts

```diff
@@ -46,12 +53,26 @@ export class WrappedExpiringStorage<TKey, TValue> implements ExpiringStorage<TKe
     }
   }
 
+  /**
+   * Deletes all entries that have expired.
+   */
+  private async cleanup(): Promise<void> {
+    const expired: string[] = [];
+    for await (const [ key, value ] of this.source.entries()) {
+      const { expires } = this.toData(value);
+      if (this.isExpired(expires)) {
+        expired.push(key);
```

Comment:
Or you could already
```suggestion
        deletions.push(this.source.delete(key));
```

Reply:
It should be fine with our current implementations, but deleting things while iterating over them always feels a bit risky to me, which is the reason why I did it like this.